src/js/common/schemaform/FormApp.jsx

```diff
@@ -40,6 +40,16 @@ class FormApp extends React.Component {
     if (window.History) {
       window.History.scrollRestoration = 'manual';
     }
+
+    // If we start in the middle of a form, redirect to the beginning
+    const currentPath = this.props.location.pathname;
+    const firstPagePath = this.props.routes[this.props.routes.length - 1].pageList[0].path;
```

Comment:
This is making an assumption that the first page is always shown, which is true now, but we might want to make a note of it here. It's possible for someone to create a form that skips the intro page and makes the first page use `depends`.

I think that's rare enough that we don't need to account for it, but I think a comment would be helpful.